beacon-chain/rpc/eth/validator/handlers.go

```diff
@@ -111,3 +110,52 @@ func (s *Server) SubmitContributionAndProofs(w http.ResponseWriter, r *http.Requ
 		}
 	}
 }
+
+// SubmitAggregateAndProofs verifies given aggregate and proofs and publishes them on appropriate gossipsub topic.
+func (s *Server) SubmitAggregateAndProofs(w http.ResponseWriter, r *http.Request) {
+	if r.Body == http.NoBody {
+		http2.HandleError(w, "No data submitted", http.StatusBadRequest)
+		return
+	}
+
+	var req SubmitAggregateAndProofsRequest
+	if err := json.NewDecoder(r.Body).Decode(&req.Data); err != nil {
```

Comment:
i remember sometimes during decode it's empty maybe there should be a nil check? though I'm not confident on the cases I've just seen it in the past. maybe a unit test around that covers

Reply:
what is empty? the body is not empty because we already checked it beforehand

Reply:
if there is a mismatch on the field name i think it populates it as empty and doesn't throw an error